protocol/messenger_communities.go

```diff
@@ -1000,65 +999,117 @@ func (m *Messenger) getAccountsToShare(addressesToReveal []string, airdropAddres
 			continue
 		}
 
-		verifiedAccount, err := m.accountsManager.GetVerifiedWalletAccount(m.settings, walletAccount.Address.Hex(), password)
+		requestID := communities.CalculateRequestID(memberPubKey, communityID)
+		msgsToSign = append(msgsToSign, account.SignParams{
+			Data:    types.EncodeHex(crypto.Keccak256(m.IdentityPublicKeyCompressed(), communityID, requestID)),
+			Address: walletAccount.Address.Hex(),
+		})
+	}
+
+	return msgsToSign, nil
+}
+
+func (m *Messenger) GenerateJoiningCommunityRequestsForSigning(memberPubKey string, communityID types.HexBytes, addressesToReveal []string) ([]account.SignParams, error) {
+	if len(communityID) == 0 {
+		return nil, errors.New("communityID has to be provided")
+	}
+	return m.generateCommunityRequestsForSigning(memberPubKey, communityID, addressesToReveal)
+}
+
+func (m *Messenger) GenerateEditCommunityRequestsForSigning(memberPubKey string, addressesToReveal []string) ([]account.SignParams, error) {
+	return m.generateCommunityRequestsForSigning(memberPubKey, []byte{}, addressesToReveal)
+}
+
+// Signs the provided messages with the provided accounts and password.
+// Provided accounts must not belong to a keypair that is migrated to a keycard.
+// Otherwise, the signing will fail, cause such accounts should be signed with a keycard.
+func (m *Messenger) SignData(signParams []account.SignParams) ([]string, error) {
+	signatures := make([]string, len(signParams))
+	for i, param := range signParams {
+		if param.Address == "" {
```

Comment:
You can add verification function to SignParams struct.

Reply:
Done.